mqtt-connect/websocket/src/main/java/com/cumulocity/mqtt/connect/client/websocket/MqttWebSocketSubscriber.java

```diff
@@ -1,53 +1,51 @@
 package com.cumulocity.mqtt.connect.client.websocket;
 
-import com.cumulocity.mqtt.connect.client.GenericMqttMessageListener;
-import com.cumulocity.mqtt.connect.client.GenericMqttSubscriber;
-import com.cumulocity.mqtt.connect.client.exception.GenericMqttClientException;
+import com.cumulocity.mqtt.connect.client.MqttClientException;
+import com.cumulocity.mqtt.connect.client.MqttMessageListener;
+import com.cumulocity.mqtt.connect.client.MqttSubscriber;
 import com.cumulocity.sdk.client.messaging.notifications.TokenApi;
 
 import java.net.URI;
 import java.util.concurrent.TimeUnit;
 
-class GenericMqttWebSocketSubscriber implements GenericMqttSubscriber {
+class MqttWebSocketSubscriber implements MqttSubscriber {
 
-    private final static String SUBSCRIBER = "javaGenericMqttWSClientSub";
+    private final static String SUBSCRIBER = "mqttConnectSubscriber";
     private final static String WEBSOCKET_URL_PATTERN = "%s/notification2/consumer/?token=%s";
 
     private final String webSocketBaseUrl;
-    private final GenericMqttWebSocketConfig config;
+    private final MqttWebSocketConfig config;
     private final TokenSupplier tokenSupplier;
 
-    private GenericMqttWebSocketClient consumer;
+    private MqttWebSocketClient consumer;
 
-    GenericMqttWebSocketSubscriber(String webSocketBaseUrl, TokenApi tokenApi, GenericMqttWebSocketConfig config) {
+    MqttWebSocketSubscriber(String webSocketBaseUrl, TokenApi tokenApi, MqttWebSocketConfig config) {
         this.webSocketBaseUrl = webSocketBaseUrl;
         this.config = config;
         this.tokenSupplier = new TokenSupplier(tokenApi, config.getTopic(), SUBSCRIBER);
```

Comment:
question: as for the producer, do all subscribers have the same name and is that an issue?

Reply:
See below answer